File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/FormContexts/Northwind.TerritoryForm.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;
    using BasicApplication;

    public partial class TerritoryForm : PrefixedContext
    {
        public TerritoryForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor TerritoryID { get { return ById<StringEditor>("TerritoryID"); } }
        public StringEditor TerritoryDescription { get { return ById<StringEditor>("TerritoryDescription"); } }
        public RegionEditor RegionID { get { return ById<RegionEditor>("RegionID"); } }
    }
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.TerritoryForm.ts
import { initFormType, LookupEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface TerritoryForm {
    TerritoryID: StringEditor;
    TerritoryDescription: StringEditor;
    RegionID: LookupEditor;
}

export class TerritoryForm extends PrefixedContext {
    static readonly formKey = 'Northwind.Territory';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!TerritoryForm.init) {
            TerritoryForm.init = true;

            initFormType(TerritoryForm, [
                'TerritoryID', StringEditor,
                'TerritoryDescription', StringEditor,
                'RegionID', LookupEditor
            ]);
        }
    }
}